pip_compile/scripts/pip-compile-container.sh

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env sh
+set -e
+
+# Docker Arguments
+IMAGE_REF="$1"
+IMAGE_CONTEXT="$2"
+DOCKERFILE="$3"
+IMAGE_TARGET="$4"
+BUILD_TAG="$IMAGE_REF:build"
+
+# Requirements Paths
+LOCAL_REQ_PATH="$5"
+REQ_MOUNT_PATH="/.pip-requirements/mount/"
+
+# Script Paths
+LOCAL_SCRIPT_PATH=$(dirname "$0")
+REQ_SCRIPT_PATH="/.pip-requirements/scripts/"
+
+# Build image.
+echo "Please wait: Building Docker image..."
+docker build --quiet \
+  --tag "$BUILD_TAG" \
+  --target "$IMAGE_TARGET" \
```

Comment:
If the Dockerfile doesn't have a target between pip-tools being installed and the full requirements being installed, a faulty requirements.txt file can cause this build process to fail as well:

```
Running cmd: sh /Users/noelle/dev/DirectEmployers/tilt-extensions/pip_compile/scripts/pip-compile-container.sh postajob /Users/noelle/dev/DirectEmployers/postajob Dockerfile backend .
Please wait: Building Docker image...
Dockerfile:58
--------------------
  57 |     COPY --chown=django requirements.txt ./
  58 | >>> RUN --mount=type=cache,target=$CACHE_DIR,uid=1000,gid=1000 \
  59 | >>>     pip install -r requirements.txt
  60 |     COPY --chown=django requirements.in ./
--------------------
ERROR: failed to solve: process "/bin/sh -c pip install -r requirements.txt" did not complete successfully: exit code: 1

View build details: docker-desktop://dashboard/build/desktop-linux/desktop-linux/7y31u6b0cf2r79mdbltpt1u6v
```

This means the "Requirements could not be compiled if requirements were in a broken state" problem isn't solved by this.

Reply:
I added an asterisk to that bullet point in the PR.